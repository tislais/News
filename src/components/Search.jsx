import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ searchTerm, onChange, onSubmit }) => {
  return (  
    <nav className="p-6 bg-indigo-400">
      <form onSubmit={onSubmit}>
        <label htmlFor="searchTerm" 
          className="text-white font-semibold">Search</label>
        <input 
          className="ml-3 rounded-sm p-2" 
          id="searchTerm" 
          type="text" 
          name="searchTerm"
          value={searchTerm}
          onChange={onChange} />
      </form>
    </nav>
  );
};

Search.propTypes = {
  searchTerm: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Search;
