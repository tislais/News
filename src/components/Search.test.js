import React from 'react';
import { render } from '@testing-library/react';
import Search from './Search';

describe('search component', () => {
  it('renders an input', () => {
    const term = 'covid';
    const { asFragment } = render(<Search search={term} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
