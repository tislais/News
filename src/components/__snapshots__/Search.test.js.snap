// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search component renders an input 1`] = `
<DocumentFragment>
  <nav
    class="p-6 bg-indigo-400"
  >
    <form>
      <label
        class="text-white font-semibold"
        for="searchTerm"
      >
        Search
      </label>
      <input
        class="ml-3 rounded-sm p-2"
        id="searchTerm"
        name="searchTerm"
        placeholder="Topic..."
        type="text"
        value=""
      />
    </form>
  </nav>
</DocumentFragment>
`;
